File: Génération_Code_Application/Opus_PureRoots/backend/app/schemas/system_log.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class SystemLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    event_type: str
    user_id: Optional[int]
    username: Optional[str]
    ip_address: Optional[str]
    success: bool
    message: str
    details: Optional[str]
    created_at: datetime


class SystemLogListResponse(BaseModel):
    items: list[SystemLogRead]
    total: int
    limit: int
    offset: int

File: Websites/Opus_PureRoots/frontend/src/types/systemLog.ts
export interface SystemLog {
  id: number;
  event_type: string;
  user_id: number | null;
  username: string | null;
  ip_address: string | null;
  success: boolean;
  message: string;
  details: string | null;
  created_at: string;
}

export interface SystemLogListResponse {
  items: SystemLog[];
  total: number;
  limit: number;
  offset: number;
}

export const EVENT_TYPE_LABELS: Record<string, string> = {
  login_success: "Login success",
  login_failure: "Login failure",
  logout: "Logout",
  register: "Registration",
  profile_update: "Profile change",
  checkout_success: "Checkout success",
  checkout_failure: "Checkout failure",
  balance_adjustment: "Balance adjustment",
  admin_user_update: "Admin user update",
};
